---
title: "PUBLIC HEALTH WHOLE GENOME SEQUENCING ANALYSIS REPORT"
output:
  html_document:
    theme: paper
    df_print: paged
---

```{css, echo=FALSE}
@import url('https://fonts.googleapis.com/css2?family=Noto+Sans+JP:wght@100;400;500;700&family=Roboto:wght@100;300;400;500;700&display=swap');
body {
  font-family: 'Noto Sans JP', Arial, Helvetica, sans-serif;
  font-size: 2rem;
  background-color: #FFF;
}

.title {
  font-weight: 300;
  text-align: center;
  margin-bottom: 50px;
  margin-top: 160px;
  font-size: 6rem !important;
}

.subtitle {
  font-weight: 300;
  text-align: center;
  font-size: 3rem !important;
}

h1, h2 {
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  margin: 0px;
  font-weight: 700;
}

h3, h4, h5, h6 {
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  margin: 10px;
  font-weight: 500;
}

.main-container{
  width: 85%;
  max-width: none;
  background-color: #FFF;
}
.section {
  margin-bottom: 50px;
}
h1 { font-size: 3.4rem;}
h2 { font-size: 3rem; }
h3 { font-size: 2.6rem; }
h4 { font-size: 2.2rem; }
h5 { font-size: 2rem; }

h2:before,
h2::after {
    display: inline-block;
    content: "";
    border-top: .3rem solid black;
    width: 4rem;
    margin: 0 1rem;
    transform: translateY(-1rem);
}

.titlePanel {
  background-color: #F0F0F0;
  position:absolute;
  top:0px;
  height: 135px;
  width: 100%;
  left:0px;
}
```

<div class ="titlePanel">
```{r, echo=FALSE, fig.align='left',out.height='100px',out.extra="style='margin:20px'"}
knitr::include_graphics("ar_logo.png")
```
</div>

```{r setup, include=FALSE}

#if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install("ggtree")

library(ggplot2)
library(plotly)
library(pander)
library(ape)
library(ggtree)

knitr::opts_chunk$set(echo = FALSE)

```

<div class="subtitle">
NAME OF THE ORGANISM / GENES OR OUTBREAK ANALYSIS REPORT
</div>

<center>
```{r intro}

# create table
df <- data.frame(date="Date",project="Project",name="Name")
colnames(df) <- c("REPORT DATE","PROJECT NAME","PREPARED BY")

# plot table
pander(df)
```
</center>

## SUMMARY

Include one sentence on the outcome of the investigation

## SAMPLES
<center>
```{r ar}

# create table
df <- data.frame('Lab ID'='Laboratory ID',
  'Isolate collection date'='Date/Of/Collection',
  'Local ID'='Local ID',
  'Species ID'='Species ID',
  'Specimen Source'= 'Specimen Source',
  'MLST'= 'MLST Result',
  'Resistance Genes'='Genes Detected',
  'Notes'='Extra Comments')

# plot table
panderOptions("table.split.table", Inf) 
pander(df)

```
</center>
## RELATEDNESS (FOR OUTBREAK REPORTS)

### HEATMAP
<center>
```{r heatmap}

# create matrix
m <- matrix(rnorm(9), nrow = 3, ncol = 3)

fig <- plot_ly(
    x = c("a", "b", "c"), y = c("d", "e", "f"),
    z = m, type = "heatmap"
)

# plot heatmap
fig

```
</center>

### TREE
<center>
```{r tree}

# create tree
n_samples <- 20
n_grp <- 4
tree <- ape::rtree(n = n_samples)

# metadata
id <- tree$tip.label
set.seed(42)
grp <- sample(LETTERS[1:n_grp], size = n_samples, replace = T)
dat <- tibble::tibble(id = id,
                      grp = grp)
# plot tree
p1 <- ggtree(tree)
metat <- p1$data %>%
  dplyr::inner_join(dat, c('label' = 'id'))
p2 <- p1 +
  geom_point(data = metat,
             aes(x = x,
                 y = y,
                 colour = grp,
                 label = id))
plotly::ggplotly(p2)

```
</center>

## ADDITIONAL ANALYSES

* AR Gene Table
* Virulence Factors
* Cluster Table

## METHODS

## DISCLAIMERS
 
Ideas: 
1. There is no set number of SNPs that classifies an outbreak. Incorporation of clinical and epidemiological data with genotypic results is recommended.
2. Identification of resistance genes has not been compared with phenotypic susceptibility testing; therefore, correlation has not been determined.
