---
title: "PUBLIC HEALTH WHOLE GENOME SEQUENCING ANALYSIS REPORT"
output:
  html_document:
    theme: paper
    df_print: paged
---

```{css, echo=FALSE}
@import url('https://fonts.googleapis.com/css2?family=Noto+Sans+JP:wght@100;400;500;700&family=Roboto:wght@100;300;400;500;700&display=swap');
body {
  font-family: 'Noto Sans JP', Arial, Helvetica, sans-serif;
  font-size: 2rem;
  background-color: #FFF;
}

.title {
  font-weight: 300;
  text-align: center;
  margin-bottom: 50px;
  margin-top: 160px;
  font-size: 6rem !important;
}

.subtitle {
  font-weight: 300;
  text-align: center;
  font-size: 3rem !important;
}

h1, h2 {
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  margin: 0px;
  font-weight: 700;
}

h3, h4, h5, h6 {
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  margin: 10px;
  font-weight: 500;
}

.main-container{
  width: 85%;
  max-width: none;
  background-color: #FFF;
}
.section {
  margin-bottom: 50px;
}
h1 { font-size: 3.4rem;}
h2 { font-size: 3rem; }
h3 { font-size: 2.6rem; }
h4 { font-size: 2.2rem; }
h5 { font-size: 2rem; }

h2:before,
h2::after {
    display: inline-block;
    content: "";
    border-top: .3rem solid;
    width: 4rem;
    margin: 0 1rem;
    transform: translateY(-1rem);
}

.titlePanel {
  background-color: #F0F0F0;
  position:absolute;
  top:0px;
  height: 135px;
  width: 100%;
  left:0px;
}
```

<div class ="titlePanel">
```{r, echo=FALSE, fig.align='left',out.height='100px',out.extra="style='margin:20px'"}
knitr::include_graphics(config$logo)
```
</div>

```{r setup, include=FALSE}

#if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install("ggtree")

library(ggplot2)
library(plotly)
library(pander)
library(kableExtra)
library(ape)
library(ggtree)
library(heatmaply)
library(phytools)

knitr::opts_chunk$set(echo = FALSE,warning=FALSE,message=FALSE)

```

<div class="subtitle">
`r subHeaderText`
</div>

<center>
```{r intro}

# create table
colnames(headerDF) <- c("REPORT DATE","PROJECT NAME","PREPARED BY")

# plot table
pander(headerDF)
```
</center>

## SUMMARY

`r summaryTEXT`

## SAMPLES
<center>
```{r samples}

# create table
colnames(sampleDF) <- c("Lab ID", "Isolate Collection Date", "Local ID", "Species ID", "Specimen Source", "MLST", "Resistance Genes", "Comments","TreeGroup")
# plot table
kbl(sampleDF[,1:8]) %>%
  kable_styling(bootstrap_options = c("hover", "condensed", "responsive"),
  fixed_thead = T,
  font_size = 14)
```
</center>
`r if(exists('snpData') | exists('treepath')){"## RELATEDNESS"}`

`r if(exists('snpData')){"### HEATMAP"}`
`r if(exists('snpData')){config$heatmap.text}`
<center>
```{r heatmap}
if(exists('snpData') && config$show.snp==TRUE){
  heatmaply(snpData,
            cellnote=snpData,
            cellnote_textposition = "middle center",
            colors = viridis(n = 256, alpha = 1, begin = 1, end = 0, option = "viridis"),
            show_dendrogram = c(FALSE, FALSE),
            dist_method=config$heat.dist.method, label_names = c("Row", "Column", "SNPs"))
}

if(exists('snpData') && config$show.snp==FALSE){
  rownames(snpData) <- snpData$X
  snpData <- snpData [,-1]
  heatmaply(snpData,
            colors = viridis(n = 256, alpha = 1, begin = 1, end = 0, option = "viridis"),
            show_dendrogram = c(FALSE, FALSE),
            dist_method=config$heat.dist.method)
}

```
</center>
`r if(exists('snpData')){"***"}`

`r if(exists('treepath')){"### PHYLOGENETIC TREE"}`
`r if(exists('treepath')){config$tree.text}`
`r if(exists('treepath') & exists('cgstats')){paste("Core Genes Identified: ",cgstats[cgstats$V1 == 'Core genes',3],'\n')}`
`r if(exists('treepath') & exists('cgstats')){paste("Total Genes Identified: ",cgstats[cgstats$V1 == 'Total genes',3])}`
<center>
```{r tree, message=FALSE}
if(exists('treepath')){
  set.seed(42)
  tree <- ape::read.tree(treepath)
  
  if(exists('cgstats')){
    paste("Core Genes Identified: ",cgstats[cgstats$V1 == 'Core genes',3])
    paste("Total Genes Identified: ",cgstats[cgstats$V1 == 'Total genes',3])
  }

  if(config$root.method == 'midpoint'){
    tree <- midpoint.root(tree)
  }
  else if(config$root.method == 'unrooted'){
    tree <- tree
  }
  else {
    node <- match(config$root.method, tree$tip.label)
    if(is.na(node)){
      message('Root sample ID not found')
      quit(save="no", status=1)
    } else {
      tree <- reroot(tree,node)
    }
  }

  # metadata
  id <- tree$tip.label

  # group samples if we have a grouping catagory
  if(any(!is.na(sampleDF$TreeGroup))){
    #reorder sampleDF
    orderedDF <- sampleDF[match(id,sampleDF$'Lab ID'),]
    grp <- orderedDF$TreeGroup
  }
  else {
    grp <- NULL
  }

  dat <- tibble::tibble(id = id)
  # plot tree
  ggtree_plot <- ggtree(tree)
  metat <- ggtree_plot$data %>%
    dplyr::inner_join(dat, c('label' = 'id'))

  ggtree_plot <- ggtree_plot +
    geom_point(data = metat,
               aes(x = x,
                   y = y,
                   colour = grp,
                   label = id, text = paste0(id)),show.legend = FALSE)


  if(config$show.bootstrap){
     #filter bootstrap values
     d <- ggtree_plot$data
     bs <- d[!d$isTip,]
     bs$label <- as.numeric(bs$label)
     bs <- bs[bs$label > config$bootstrap.threshold,]
     ggtree_plot <- ggtree_plot + geom_text(data=bs,aes(label=label,),position = position_dodge2(width = 0.001))
  }

  ggplotly(ggtree_plot, tooltip = "text")
}
```
</center>
`r if(exists('treepath')){"***"}`

`r if(exists('ar_summary')){"## AR GENE SUMMARY"}`
`r if(exists('ar_summary')){config$ar_gene_summary.text}`
<center>
```{r ar_table}
if(exists('ar_summary')){
  ar_summary <- ar_summary[order(ar_summary$Sample),]
  row.names(ar_summary) <- NULL
  # plot table
  kbl(ar_summary) %>%
    kable_styling(bootstrap_options = c("hover", "condensed", "responsive"),
    fixed_thead = T,
    font_size = 14) %>%
    pack_rows(index=table(ar_summary$Sample))
}
```
</center>

`r if(exists('optionalData')){"## ADDITIONAL ANALYSES"}`
`r if(exists('optionalData')){config$optionaldata.text}`
<center>
```{r additional tables, results='asis'}
if(exists('optionalData')){
  for(i in 1:length(optionalData)){
    print(kbl(optionalData[[i]]) %>%
      kable_styling(bootstrap_options = c("hover", "condensed", "responsive"),
      fixed_thead = T,
      font_size = 14))
    cat('\n')
  }
}
```
</center>

## METHODS
`r methodsTEXT`
## DISCLAIMER
`r disclaimerTEXT`
